src/report/mermaid.js

```diff
@@ -83,40 +99,71 @@ const convertedSubgraphSources = (pCruiseResult) => {
   return lTree;
 };
 
-const focusHighlights = (pModules) => {
+/**
+ * @param {import("../../types/cruise-result").IModule[]} pModules
+ * @param {Map<string, string>} pMinifiedNames
+ */
+const focusHighlights = (pModules, pMinifiedNames) => {
   const lHighLightStyle = "fill:lime,color:black";
 
   return pModules
     .filter((pModule) => pModule.matchesFocus === true)
-    .reduce(
-      (pAll, pModule) =>
-        (pAll += `\nstyle ${hashNodeName(pModule.source)} ${lHighLightStyle}`),
-      ""
-    );
+    .reduce((pAll, pModule) => {
+      const lSource = pMinifiedNames.get(pModule.source) || pModule.source;
+      return (pAll += `\nstyle ${hashNodeName(lSource)} ${lHighLightStyle}`);
+    }, "");
 };
 
-const renderMermaidThing = (pCruiseResult) => {
-  const subgraphs = convertedSubgraphSources(pCruiseResult);
-  const edges = convertedEdgeSources(pCruiseResult);
+/**
+ * @param {import("../../types/cruise-result").IModule[]} pModules
+ * @param {Boolean} pMinify
+ * @return {Map<string, string>}
+ */
+const minifiedNames = (pModules, pMinify) => {
+  const names = new Map();
+  if (!pMinify) return names;
+
+  let lCount = 0;
+  pModules.forEach((pModule) => {
+    const lPaths = pModule.source.split("/");
+
+    for (let lIndex = 0; lIndex < lPaths.length; lIndex += 1) {
+      const lName = lPaths.slice(0, lIndex + 1).join("/");
```

Comment:
(You may already know this since you have been working on the source, but I answer as documentation that will remain in the future.)

> What is the reason for splitting, and later joining the source string?

The file path of source must be nested and represented as a mermaid subgraph. To express that multiple files exist in the same level, a name  for each levels is added to the hash key.  That is, if the following two files exist

- /foo/bar/baz.js
- /foo/bar/qux.js

The name that should be registered as a hash key at this time looks like this:

- "/foo" 
- "/foo/bar"
- "/foo/bar/baz.js"
- "/foo/bar/qux.js"

And as for the syntax of mermaid, the output is as follows.

```mermaid
flowchart LR

subgraph 0["/foo"]
  subgraph 1["/foo/bar"]
  2["/foo/bar/baz.js"]
  3["/foo/bar/qux.js"]
  end
end
```

To achieve this, each level is splitted and joined.